app/models/cattle.rb

```diff
@@ -24,6 +25,48 @@ def generate_tag
     "#{country_code}#{herdmark}#{check_digit}#{format('%05d', individual_number)}"
   end
 
+  def imprint(data)
+    imprint = biometric_imprint.create(image_uri: 'temporary')
+    image_uri = "cattle/#{user.id}/#{id}/#{imprint.id}-imprint-original"
+    $s3.put_object(
+      acl: 'private',
+      body: data,
+      bucket: 'cowhub-production-images',
+      key: image_uri
+    )
+    imprint.image_uri = image_uri
+    imprint.save
+    imprint
+  end
+
+  def add_image(data)
+    profile = profile_picture.create(image_uri: 'temporary')
+    image_uri = "cattle/#{user.id}/#{id}/#{profile.id}-profile-original"
+    $s3.put_object(
+      acl: 'private',
+      body: data,
+      bucket: 'cowhub-production-images',
+      key: image_uri
+    )
+    profile.image_uri = image_uri
+    profile.save
+    profile
+  end
+
+  def images
```

Comment:
You need to split this into two functions, `biometric_imprint_images` and `profile_images`